### ISSUE TITLE
Layout is invisible if first item is gone

### ISSUE THREAD

**Issue by bashtian** [google/flexbox-layout#283]
I have a layout with multiple children and when I set the first item to View.GONE the whole layout is invisible. If I only set the second item to gone the layout works fine.

There seems to be a problem with calculating flexLine.mItemCount, in this line [0] it's always 1 when the first item is gone.

[0] https://github.com/google/flexbox-layout/blob/dev_recyclerview/flexbox/src/main/java/com/google/android/flexbox/FlexboxLayout.java#L658

**Comment by thagikura** [comment]
Thanks for reporting. It's likely a bug with FlexboxLayout

**Comment by thagikura** [comment]
Hmm, 

I tried to reproduce it on my end, but couldn't reproduce it so far.
I changed visibility of the first item in the [fragment_flexboxlayout.xml](https://github.com/google/flexbox-layout/blob/dev_recyclerview/demo-playground/src/main/res/layout/fragment_flexboxlayout.xml) as gone, but the whole layout was still visible in the demo-playground app.

Is it possible to share your reproducible code?

```xml
<com.google.android.flexbox.FlexboxLayout xmlns:android="http://schemas.android.com/apk/res/android"
    xmlns:app="http://schemas.android.com/apk/res-auto"
    xmlns:tools="http://schemas.android.com/tools"
    android:id="@+id/flexbox_layout"
    android:layout_width="match_parent"
    android:layout_height="match_parent"
    app:alignContent="flex_start"
    app:alignItems="flex_start"
    app:flexWrap="wrap"
    tools:showIn="@layout/activity_main">

    <TextView
        android:id="@+id/textview1"
        style="@style/FlexItem"
        android:layout_width="@dimen/flex_item_length2"
        android:layout_height="@dimen/flex_item_length"
        android:text="@string/one"
        android:visibility="gone"
    />

    <TextView
        android:id="@+id/textview2"
        style="@style/FlexItem"
        android:layout_width="@dimen/flex_item_length3"
        android:layout_height="@dimen/flex_item_length"
        android:text="@string/two" />

    <TextView
        android:id="@+id/textview3"
        style="@style/FlexItem"
        android:layout_width="@dimen/flex_item_length"
        android:layout_height="@dimen/flex_item_length"
        android:text="@string/three" />

</com.google.android.flexbox.FlexboxLayout>
```

Thanks.

**Comment by bashtian** [comment]
I changed the visibility programmatically from visible to gone on the first item, maybe that could be the problem.

**Comment by thagikura** [comment]
Hmm,

I change the visibility programmatically but the same result.

I changed the OnClickListener in the [FlexboxLayoutFragment](https://github.com/google/flexbox-layout/blob/dev_recyclerview/demo-playground/src/main/java/com/google/android/flexbox/FlexboxLayoutFragment.java)
in line 82 as follows (changing the first item's visibility as GONE at the bottom).

```java
      addFab.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View view) {
                    int viewIndex = mFlexContainer.getFlexItemCount();
                    // index starts from 0. New View's index is N if N views ([0, 1, 2, ... N-1])
                    // exist.
                    TextView textView = createBaseFlexItemTextView(activity, viewIndex);
                    FlexboxLayout.LayoutParams lp = new FlexboxLayout.LayoutParams(
                            ViewGroup.LayoutParams.WRAP_CONTENT,
                            ViewGroup.LayoutParams.WRAP_CONTENT);
                    fragmentHelper.setFlexItemAttributes(lp);
                    textView.setLayoutParams(lp);
                    textView.setOnClickListener(new FlexItemClickListener(activity,
                            new FlexItemChangedListenerImpl(mFlexContainer), viewIndex));
                    mFlexContainer.addView(textView);
                    mFlexContainer.getFlexItemAt(0).setVisibility(View.GONE);
                }
            });
```

**Comment by bashtian** [comment]
Here is a layout where the first item is set to GONE and the last item also disappears.
```

    <com.google.android.flexbox.FlexboxLayout
        android:id="@+id/flexbox_layout"
        android:layout_width="match_parent"
        android:layout_height="match_parent"
        app:alignContent="flex_start"
        app:alignItems="flex_start"
        app:flexWrap="wrap"
        tools:showIn="@layout/activity_main">

        <TextView
            android:id="@+id/textview1"
            android:layout_width="wrap_content"
            android:layout_height="wrap_content"
            android:text="1"
            android:visibility="gone"
            app:layout_wrapBefore="true"
            />

        <TextView
            android:id="@+id/textview2"
            android:layout_width="wrap_content"
            android:layout_height="wrap_content"
            android:text="2"
            app:layout_wrapBefore="true"
            />

        <TextView
            android:id="@+id/textview3"
            android:layout_width="wrap_content"
            android:layout_height="wrap_content"
            android:text="3"
            app:layout_wrapBefore="true"
            />

    </com.google.android.flexbox.FlexboxLayout>
```
Maybe the disappearing of the whole layout is related to this bug.

**Comment by thagikura** [comment]
Thanks for pasting the example layout.
The issue happened in the case where the first view's visibility is gone and the second view's `layout_wrapBefore` is set to true.

In that case, the second view's position is misplaced.
Sent #300 to fix this. The fix will be submitted to the `dev_recyclerview` it will be fixed in the master once the dev_recyclerview branch is merged to master.

**Comment by thagikura** [comment]
Should be fixed by #300

**Comment by bashtian** [comment]
Thank you for the fix. I found another issue that seems to be related. I created a new ticket https://github.com/google/flexbox-layout/issues/303
